components/frontend/src/App.js

```diff
@@ -33,15 +33,19 @@ class App extends Component {
         const pathname = this.history.location.pathname;
         const report_uuid = pathname.slice(1, pathname.length);
         const report_date_iso_string = registeredURLSearchParams(this.history).get("report_date") || "";
-        const report_date_string = isValidDate_YYYYMMDD(report_date_iso_string) ? report_date_iso_string.split("-").reverse().join("-") : "";
+        let reportDate = null;
+        if (isValidDate_YYYYMMDD(report_date_iso_string)) {
+            reportDate = new Date(report_date_iso_string);
+            reportDate.setHours(23, 59, 59);
```

Comment:
Why 1 second before midnight instead of just 0,0,0 ?

Reply:
Oh, nvm, this was old code that's been moved. Is it tested elsewhere? Because in this PR I don't see any test on 23:59:59

Reply:
It's one second before midnight because on the server the measurement end timestamp is compared with the report date and we want measurements that end on the report date to be included. Calling `setHours(0, 0, 0)` would effectively set the report date to the day before.